### PR TITLE
Adding Thales NShield integration tests and validation

### PR DESCRIPTION
**Pull request by nickrmc83**
The changes include vendor defined mechanisms (see vendor/ThalesNShield.json). All test

### CODE REVIEW

**Comment by rmhrisk** [comment]
@nickrmc83 thanks for this PR, we will review the branch conflicts and merge this as soon as possible.

File: build/console/console.js [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env node
```
Comment (microshine) [inline build/console/console.js]: Current line is required to register graphene module like CLI

**Comment by rmhrisk** [comment]
@nickrmc83 we believe we have aligned your PR with the current code base, can you re-run the tests in your environment to confirm it still works as expected before we merge?

**Comment by nickrmc83** [comment]
I'll give it a go this morning.

Nick

**Comment by nickrmc83** [comment]
I had to make the below changes to make things work as expected.
`
diff --git a/src/keys/rsa/pss.ts b/src/keys/rsa/pss.ts
index 1de4886..41320ce 100644
--- a/src/keys/rsa/pss.ts
+++ b/src/keys/rsa/pss.ts
@@ -28,7 +28,7 @@ export class RsaPssParams implements IParams {
 
     type = MechParams.RsaPSS;
 
-    constructor(hashAlg: MechanismEnum = MechanismEnum.SHA1, mgf: RsaMgf = RsaMgf.MGF1_SHA1, saltLen: number = 0) {
+    constructor(hashAlg: MechanismEnum = MechanismEnum.SHA1, mgf: RsaMgf = RsaMgf.MGF1_SHA1, saltLen: number = 20) {
         this.hashAlgorithm = hashAlg;
         this.mgf = mgf;
         this.saltLength = saltLen;
diff --git a/test/session.js b/test/session.js
index f3288d5..f1123c0 100644
--- a/test/session.js
+++ b/test/session.js
@@ -43,7 +43,7 @@ describe("Session", function() {
         session.close();
     });
 
-function changePIN(session, userType, oldPIN, newPIN) {
+    function changePIN(session, userType, oldPIN, newPIN) {
         session.login(oldPIN, userType);
         session.setPin(oldPIN, newPIN);
         session.logout();
@@ -63,7 +63,7 @@ function changePIN(session, userType, oldPIN, newPIN) {
             throw e;
         }
         session.close();
-    });
+    }).timeout(60000);
     
     it("changing PIN for SO", function() {
         var session = slot.open(2|4);
@@ -79,7 +79,7 @@ function changePIN(session, userType, oldPIN, newPIN) {
             throw e;
         }
         session.close();
-    });
+    }).timeout(60000);
 
     it("create", function() {
         // create new session for current test
@@ -127,7 +127,8 @@ function changePIN(session, userType, oldPIN, newPIN) {
             valueLen: 256/8,
             extractable: false,
             sensitive: false,
-            encrypt: true
+            encrypt: true,
+            private: true
         });
 
         // test objs length
`

**Comment by rmhrisk** [comment]
Superseded

**Pull request closed without merging**